packages/drivers/odsp-driver/src/createNewUtils.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { v4 as uuid } from "uuid";
+import { ISummaryTree, SummaryType } from "@fluidframework/protocol-definitions";
+import { fromUtf8ToBase64, Uint8ArrayToString } from "@fluidframework/common-utils";
+import { IBlob, IOdspSnapshot, ITree, ITreeEntry } from "./contracts";
+
+/**
+ * Converts a summary(ISummaryTree) taken in detached container to IOdspSnapshot tree
+ */
+export function convertSummaryTreeToIOdspSnapshot(summary: ISummaryTree): IOdspSnapshot {
+    const trees: ITree[] = [];
+    const blobs: IBlob[] = [];
+    const mainTree: ITree = {
+        entries: [],
+        id: uuid(),
+        sequenceNumber: 0,
+    };
+    trees.push(mainTree);
+    const snapshotTree: IOdspSnapshot = {
+        trees,
+        blobs,
+        id: mainTree.id,
+    };
+
+    convertSummaryTreeToIOdspSnapshotCore(summary, mainTree.entries, blobs);
```

Comment:
I feel like it's a bit more readable:

```typescript
    const snapshotTree: IOdspSnapshot = {
        trees: [ {
           entries: [],
           id: uuid(),
           sequenceNumber: 0,
        }],
        blobs: [],
        id: mainTree.id,
    };

convertSummaryTreeToIOdspSnapshotCore(summary, snapshotTree.trees.entrees, snapshotTree.blobs);
```

Or actually this feels even better:
```
const {entrees, blobs} = convertSummaryTreeToIOdspSnapshotCore(summary);
    const snapshotTree: IOdspSnapshot = {
        trees: [ {
           entries,
           id: uuid(),
           sequenceNumber: 0,
        }],
        blobs,
        id: mainTree.id,
    };
```

Reply:
I have changed to first suggestion as that feels natural as like we are already creating and sending to be filled by core function because it is a flat list and not hierarchical.